go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -628,7 +632,7 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 			MAC: podMac,
 		}
 		var nodeSubnets []*net.IPNet
-		if nodeSubnets = bnc.lsManager.GetSwitchSubnets(switchName); nodeSubnets == nil {
+		if nodeSubnets = bnc.lsManager.GetSwitchSubnets(switchName); nodeSubnets == nil && bnc.doesNetworkRequireIPAM() {
```

Comment:
note that releaseIPs is set to true and if there is any error, the defered ReleaseIPs() will be called for no-ipam logial_switch_port. better to change that defered function.

Reply:
I can, but the [releaseIPs](https://github.com/ovn-org/ovn-kubernetes/blob/8f78d1f58c0d762ee880e97067ff448d77decd45/go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go#L350) function is pretty harmless IMO: it'll just iterate the assigned IPs (there would be none), and do nothing right ? 

I would - at most - update the log in the deferred cleanup `else` clause when there is nothing to clean - *but* I think what is present there is good enough for the scenario where there are no assigned IPs.